src/commands/push.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  loadAPICredentials,
+} from './../auth';
+
+import {
+  DOT,
+  DOTFILE,
+} from './../dotfile';
+
+import {
+  fetchProject,
+  pushFiles,
+} from './../files';
+
+import {
+  isValidManifest,
+} from './../manifest';
+
+import {
+  checkIfOnline,
+  getProjectSettings,
+  LOG,
+  PROJECT_MANIFEST_BASENAME,
+  PROJECT_MANIFEST_FILENAME,
+  spinner,
+} from './../utils';
+
+import { readFileSync } from 'fs';
+import * as path from 'path';
+import { watchTree } from 'watch';
+import multimatch = require('multimatch');
+const inquirer = require('inquirer');
+const prompt = inquirer.prompt;
+inquirer.registerPrompt('autocomplete', require('inquirer-autocomplete-prompt'));
+
+/**
+ * Uploads all files into the script.google.com filesystem.
+ * TODO: Only push the specific files that changed (rather than all files).
+ * @param cmd.watch {boolean} If true, runs `clasp push` when any local file changes. Exit with ^C.
+ */
+export default async (cmd: { watch: boolean, force: boolean }) => {
```

Comment:
Was there any changes made to the actual code, or was it just copied and pasted as-is? Hard to tell with the diff view.

Reply:
Yes, great question.

The constants `manifestHasChanges` and `confirmManifestUpdate` were extracted to the bottom of the file and comments were added to them.

My thought is that it improves code readability. Additionally, if we wanted in the future, we could export these and test them individually.